mods/beds/api.lua

```diff
@@ -61,9 +61,11 @@ function beds.register_bed(name, def)
 				pos = pointed_thing.above
 			end
 
-			if minetest.is_protected(pos, placer:get_player_name()) and
-					not minetest.check_player_privs(placer, "protection_bypass") then
-				minetest.record_protection_violation(pos, placer:get_player_name())
+			local player_name = placer and placer:get_player_name() or ""
+
+			if minetest.is_protected(pos, player_name) and
+					not minetest.check_player_privs(player_name, "protection_bypass") then
```

Comment:
check_player_privs takes `placer`, not `placer:get_player_name()`

Reply:
How about "[It takes both](https://github.com/minetest/minetest/blob/9fa78b7/builtin/game/misc.lua#L11)" ?

Reply:
that's both intuitive and unintuitive at once..

Reply:
it's better to use placer here, rather than the name as you have the former alread

Reply:
`placer` can be nil, so `check_player_privs` will throw an error about an invalid argument type.

Reply:
is `check_player_privs("", ...")` valid?

Reply:
Yes, the alternative would have been to use `placer or ""`.

Reply:
@sfan5 That's what was used before placer could be nil I think.